src/client/renderer/r_types.h

```diff
@@ -275,12 +374,11 @@ typedef struct {
 } r_bsp_texinfo_t;
 
 typedef struct {
-	vec3_t origin;
 	vec_t radius;
-	const r_image_t *image;
-	vec3_t color;
 	uint32_t time;
 	vec_t alpha;
+
+	r_particle_t particle;
 } r_bsp_flare_t;
```

Comment:
Oh, I see. I thought the `alpha` field here was the final, "ramped" alpha after the visibility check. You already moved that into the particle. Great.